duckietown/duckiebot-interface
# enable wheel encoder to read phase for DB21M

## Description
Make Jetson Nano based Duckiebots directly use phase from wheel encoders to determine rotation direction.

## How to test
* clone the repo, switch to `daffy-wheel-encoder-phase` branch
* `cd [path/to/]dt-duckiebot-interface`
* Build: `dts devel build -f --pull -H [HOSTNAME]`
* Run: 
 * stop `duckiebot-interface` container from portainer
 * `dts devel run -H [HOSTNAME] -M -s -f -- --privileged -v /tmp/argus_socket:/tmp/argus_socket`
* Check: 
 * `dts start_gui_tools [HOSTNAME]`
 * `rostopic echo /[HOSTNAME]/left_wheel_encoder_node/tick` and manually turn the left wheel backward (ticks go down) or forward (ticks go up).

## Type of change

Please mark the relevant options with X:

- [ ] This change is constrained only to one package, any interfaces to nodes from other packages are maintained
- [ ] This change is constrained only to packages in this repository, any interfaces to nodes from other repositories are maintained
- [ ] This change is only a documentation update
- [ ] This change restructures the repository (at level higher than package level)
- [ ] This change requires a change in the Duckiebook (if yes, please describe and, if possible, make a pull request for an edit in the book and link it here)

## Style compliance
Please make sure that all your changes conform to the Duckietown (code style)[link] and code (documentation style)[link] guidelines:

- [x] I checked that this pull request is compliant with the Duckietown code style
- [x] I checked that this pull request is compliant with the Duckietown code documentation style

## Next steps
Please describe if you have any questions, comments, or doubts towards the Duckietown software guardians regarding implementation, code style, running tests, documentation, etc.

## FOR THE REVIEWERS
__IMPORTANT:__ DO NOT CHANGE THIS SECTION!

- [ ] Functionality (is the change necessary and is it implemented in the most appropriate way): @liampaull or @afdaniele
- [ ] Integration testing (verify the tests and run additional if necessary): @?
- [ ] Code style compliance: @gibernas
- [ ] Documentation compliance: @aleksandarpetrov

_To the reviewers:_ If you deem necessary, request code edits, more tests, documentation update, or additional reviewing. __Do not review your own code!__

_PR Template, last edit 23 Aug 2019_
